### PR TITLE
feat: clear local storage

### PR DESCRIPTION
**Pull request by Zfinix**
This creates methods to clear data from `SecureDataStore`, `KeyInfoStorage`  and `PlainDataStore`

### CODE REVIEW

File: packages/transaction_signing_gateway/lib/storage/cosmos_key_info_storage.dart [hunk 1]
```diff
@@ -233,4 +233,12 @@ class CosmosKeyInfoStorage implements KeyInfoStorage {
       );
     }
   }
+
+  @override
+  Future<Either<CredentialsStorageFailure, bool>> clearCredentials() async {
+    final _secureCleared = await _secureDataStore.clearAllData();
+    final _plainCleared = await _plainDataStore.clearAllData();
+    _secureCleared.fold(left, right);
+    return _plainCleared.fold(left, right);
```
Comment (wal33d006) [inline packages/transaction_signing_gateway/lib/storage/cosmos_key_info_storage.dart:242]: Secure and plain data clearing should be in separate methods both returning `Future<Either<CredentialsStorageFailure, bool>>` and both of them separately being called in this `clearCredentials` function.

File: packages/transaction_signing_gateway/lib/storage/biometric_data_store.dart [hunk 2]
```diff
@@ -124,6 +124,22 @@ class BiometricDataStore implements SecureDataStore {
...
+      final biometricStorage = BiometricStorage();
+      final storageFile = await biometricStorage.getStorage(
+        storageFileName,
+        options: _storageFileInitOptions,
+        promptInfo: promptInfo,
+      );
+      await storageFile.delete();
+      return right(true);
+    } catch (e) {
+      return right(false);
```
Comment (wal33d006) [inline packages/transaction_signing_gateway/lib/storage/biometric_data_store.dart:140]: I think this needs to be 

```
catch(ex, stack) {
return left(CredentialsStorageFailure(
              'Error while clearing data in Biometric storage',
              cause: ex,
              stack: stack,
            ))
}
```

File: packages/transaction_signing_gateway/test/mocks/test_memory_store.dart [hunk 3]
```diff
@@ -4,6 +4,12 @@ import 'package:dartz/dartz.dart';
 class TestMemoryStore implements PlainDataStore, SecureDataStore {
   final Map<String, String?> _values = {};
 
+  @override
+  Future<Either<CredentialsStorageFailure, bool>> clearAllData() async {
+    _values.clear();
```
Comment (wal33d006) [inline packages/transaction_signing_gateway/test/mocks/test_memory_store.dart:9]: Should be wrapped in try catch

File: packages/transaction_signing_gateway/lib/storage/cosmos_key_info_storage.dart (hunk 1 above)
Comment (wal33d006) [reply packages/transaction_signing_gateway/lib/storage/cosmos_key_info_storage.dart:242]: Also, this should be wrapped in try catch block.

**Review by wal33d006 — CHANGES_REQUESTED** [review]
Awesome!
Just a few minor comments!

File: packages/transaction_signing_gateway/test/mocks/test_memory_store.dart (hunk 3 above)
Comment (Zfinix) [reply packages/transaction_signing_gateway/test/mocks/test_memory_store.dart:9]: will do

File: packages/transaction_signing_gateway/lib/storage/cosmos_key_info_storage.dart (hunk 1 above)
Comment (andrzejchm) [inline packages/transaction_signing_gateway/lib/storage/cosmos_key_info_storage.dart:242]:
```suggestion
    return _secureDataStore.clearAllData() //
        .flatMap((_) => _plainDataStore.clearAllData());
```

File: packages/transaction_signing_gateway/lib/storage/flutter_secure_storage_data_store.dart [hunk 4]
```diff
@@ -52,4 +52,24 @@ class FlutterSecureStorageDataStore implements SecureDataStore {
       );
     }
   }
+
+  @override
+  Future<Either<CredentialsStorageFailure, bool>> clearAllData() async {
+    try {
+      await _store.deleteAll(
+        iOptions: const IOSOptions(accessibility: IOSAccessibility.passcode),
```
Comment (andrzejchm) [inline packages/transaction_signing_gateway/lib/storage/flutter_secure_storage_data_store.dart:60]: let's extract the IOSOptions and AndroidOptions to private static const fields inside this class to keep it DRY

File: packages/transaction_signing_gateway/test/mocks/test_memory_store.dart (hunk 3 above)
Comment (andrzejchm) [reply packages/transaction_signing_gateway/test/mocks/test_memory_store.dart:9]: actually, no need to wrap it as `clear` does not throw anything

**Review by andrzejchm — CHANGES_REQUESTED** [review]
good work! just few minor comments from my side

File: packages/transaction_signing_gateway/lib/storage/flutter_secure_storage_data_store.dart (hunk 4 above)
Comment (Zfinix) [reply packages/transaction_signing_gateway/lib/storage/flutter_secure_storage_data_store.dart:60]: will do

File: packages/transaction_signing_gateway/lib/storage/biometric_data_store.dart (hunk 2 above)
Comment (Zfinix) [reply packages/transaction_signing_gateway/lib/storage/biometric_data_store.dart:140]: Fixed

**Comment by Zfinix** [comment]
This is finally ready to be merged @andrzejchm @wal33d006

**Comment by andrzejchm** [comment]
love it!

**Review by andrzejchm — APPROVED** [review]

**Review by wal33d006 — APPROVED** [review]

**Pull request merged**